text/0000-returns-twice.md

```diff
@@ -0,0 +1,106 @@
+- Feature Name: ffi-returns-twice
+- Start Date: 2019.02.08
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC adds a new function attribute, `#[ffi_returns_twice]`, which indicates
+that an foreign function can return multiple times.
+
+# Motivation
+[motivation]: #motivation
+
+Rust assumes that function calls like:
+
+```rust
+let x = foo();
+```
+
+return only once. That is, when the execution arrives at the function call, the
+function is called, it returns a single value, and that's it. This assumption
+allows Rust to perform many optimizations.
```

Comment:
Could you add an example? I know there was one in some prior discussion that showed the need for this attribute quite convincingly.